framework/test/unit/modules/interoperability/mainchain/commands/sidechain_registration.spec.ts

```diff
@@ -0,0 +1,504 @@
+			sidechainRegistrationCommand['getInteroperabilityStore'] = jest
+				.fn()
+				.mockReturnValue({ sendInternal });
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA)
+				.mockReturnValue(chainSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHANNEL_DATA)
+				.mockReturnValue(channelSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_VALIDATORS)
+				.mockReturnValue(validatorsSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_OUTBOX_ROOT)
+				.mockReturnValue(outboxRootSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_REGISTERED_NAMES)
+				.mockReturnValue(registeredNamesSubstore);
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_REGISTERED_NETWORK_IDS)
+				.mockReturnValue(registeredNetworkIDsSubstore);
+		});
+
+		it('should add an entry to chain account substore', async () => {
+			// Arrange
+			const expectedValue = {
+				name: 'sidechain',
+				networkID,
+				lastCertificate: {
+					height: 0,
+					timestamp: 0,
+					stateRoot: EMPTY_HASH,
+					validatorsHash: computeValidatorsHash(params.initValidators, params.certificateThreshold),
+				},
+				status: CCM_STATUS_OK,
+			};
+
+			// Act
+			await sidechainRegistrationCommand.execute(context);
+
+			// Assert
+			expect(chainSubstore.setWithSchema).toHaveBeenCalledWith(
+				newChainID,
+				expectedValue,
+				chainAccountSchema,
+			);
+		});
+
+		it('should throw error if no entries found in chain account substore', async () => {
+			// Arrange
+			when(mockGetStore)
+				.calledWith(MODULE_ID_INTEROPERABILITY, STORE_PREFIX_CHAIN_DATA)
+				.mockReturnValue({
+					getWithSchema: jest.fn().mockResolvedValue(chainAccount),
+					setWithSchema: jest.fn(),
+					iterate: jest.fn().mockResolvedValue([]),
+				});
+
+			// Act
+			// Assert
+			await expect(sidechainRegistrationCommand.execute(context)).rejects.toThrow(
+				'No existing entries found in chain store',
+			);
+		});
+
+		it('should add an entry to channel account substore', async () => {
+			// Arrange
+			const expectedValue = {
+				inbox: { root: EMPTY_HASH, appendPath: [], size: 0 },
+				outbox: { root: EMPTY_HASH, appendPath: [], size: 0 },
+				partnerChainOutboxRoot: EMPTY_HASH,
+				messageFeeTokenID: { chainID: 1, localID: 0 },
+			};
+
+			// Act
+			await sidechainRegistrationCommand.execute(context);
+
+			// Assert
+			expect(channelSubstore.setWithSchema).toHaveBeenCalledWith(
+				newChainID,
+				expectedValue,
+				channelSchema,
+			);
+		});
+
+		it('should call sendInternal with a registration ccm', async () => {
+			// Act
+			await sidechainRegistrationCommand.execute(context);
+
+			// Assert
+			// Due to `timestamp` difference for input object on test run between execution and expectation, we only checking that it was called
```

Comment:
once in the source code, Data.now() is removed, this can be updated. (module code should never be non-deterministic).
If we use random for test, you could also check like
```
expect(sendInternal).toHaveBeenCalledWith({
  nonce: BigInt(0),
  otherProperties...,
  timestamp: expect.Any(Number),
});
```